### PR TITLE
feat: gray out meta data object when using `--singleLine=true`

### LINKED ISSUE #160: Gray out meta information when using --singleLine option when coloring is enabled
I really like the single-line output introduced in https://github.com/pinojs/pino-pretty/issues/97 🙇🏼🙏🏼 

image

I wonder if we could gray out the meta information by default, when coloring is enabled?

image

I think that makes the logs more readable. 

If you'd like that change, I'd appreciate a pointer on how the change would be done. I looked into `lib/utils.js` but I'm not sure how that change would be implemented, other than requiring `chalk` in this file and making this change

```diff
-       result += stringifySafe(plain)
+       result += chalk.gray(stringifySafe(plain)
```

### PR DESCRIPTION
**Pull request by gr2m**
fixes #160 

Before 

image

After

image

### CODE REVIEW

**Comment by gr2m** [comment]
I am confused, the tests should not have passed with the changes from 6641ce0, I only pushed the tests changes to show that they are failing, before pushing the fix

**Comment by gr2m** [comment]
Is it possible that on CI the colors are disabled, so testing for colors is impossible?

**Comment by gr2m** [comment]
When running this locally I get

image

image

**Comment by gr2m** [comment]
I'm not sure if this only affects my code, but it looks like tests for colors in stdout won't work in CI because colors are disabled by default. If the code changes from this PR were removed in future, the tests would still pass in CI (but fail locally)

**Comment by mcollina** [comment]
You should use https://github.com/pinojs/pino-pretty/blob/master/lib/colors.js and not chalk directly. You'll be able to test with that.

**Comment by gr2m** [comment]
I'm sorry but I don't get how to use the `colorizer` API from `colors.js` to color the meta object output as gray (debug). Can you give me a pointer plaese

**Comment by mcollina** [comment]
The `new chalk.Instance({ level })` API in https://github.com/pinojs/pino-pretty/blob/657b7ba081d1d8d8cc9fdb90d51c84a75c88d5f8/lib/colors.js#L18 will print out colors even if it's disabled in the TTY.

You will need to use `colorizer` with the `level` argument of `'trace'` (which is grey). The tricky part is that you'd need to percolate down the current colorizer from the caller function

File: lib/utils.js [hunk 1]
```diff
@@ -3,6 +3,7 @@
 const dateformat = require('dateformat')
 const stringifySafe = require('fast-safe-stringify')
 const defaultColorizer = require('./colors')()
+const chalk = require('chalk')
```
Comment (jsumners) [inline lib/utils.js:6]: As Matteo said, we have a standardized colorizer in this module. It is defined on line 5 immediately before this new import. Please use that colorizer to add color to output.

**Comment by gr2m** [comment]
Sorry I've spent another 15minutes on this I just don't know what to do with the `colorizer` argument. `colorizer` has a single method `.message()` which does not accept a level parameter. And calling colorizer as a function only returns the name of the level, not a chalk-compatible instance, e.g. `colorizer('trace') // TRACE`

Do you want to finish this up yourself? I'll learn from your changes for the next time

**Comment by jsumners** [comment]
> Sorry I've spent another 15minutes on this I just don't know what to do with the `colorizer` argument. `colorizer` has a single method `.message()` which does not accept a level parameter. And calling colorizer as a function only returns the name of the level, not a chalk-compatible instance, e.g. `colorizer('trace') // TRACE`
> 
> Do you want to finish this up yourself? I'll learn from your changes for the next time

https://runkit.com/embed/op9ez5z3pgxm

`lib/colors` returns a factory function that asks "do you want colors or not?". In either case, it returns a new function that will return a colorized level label based upon the level given, e.g. `colorize('trace')` or `colorize('info')`. This function _also_ includes a `message` method that will colorize a message according to our standardized message color: `colorizer.message('a message')`.

What you want to do is add a new method that will colorize according to a different color, e.g. `colorizer.greyMessage = (msg) => colorizeLevel('trace', msg)`.

**Comment by gr2m** [comment]
> What you want to do is add a new method that will colorize according to a different color, e.g. `colorizer.greyMessage = (msg) => colorizeLevel('trace', msg)`.

I would add that code to `lib/colors.js` as part of the factory function?

**Comment by jsumners** [comment]
> I would add that code to `lib/colors.js` as part of the factory function?

I would expect to see it defined for each colorizer type in the same general location as the definition of `colorizer.message`.

**Comment by gr2m** [comment]
see my commit at https://github.com/pinojs/pino-pretty/pull/161/commits/2d7ca36935d7cf574b39278d63023528c77992f4

That makes the tests fail locally now. How do I test for the color output?

**Comment by jsumners** [comment]
> That makes the tests fail locally now. How do I test for the color output?

https://github.com/pinojs/pino-pretty/blob/657b7ba081d1d8d8cc9fdb90d51c84a75c88d5f8/test/lib/colors.test.js#L42

**Comment by gr2m** [comment]
okay so I can write a unit test for the `colorizer.greyMessage` method, but I cannot test the color of the log output in basic.test.js?

**Comment by jsumners** [comment]
> okay so I can write a unit test for the `colorizer.greyMessage` method, but I cannot test the color of the log output in basic.test.js?

Is there a need to? If the colorizer is working as tested, then the final result is going to be correct, yes?

**Comment by gr2m** [comment]
okay, just wanted to make sure. I think it would be better to have some integration tests, because in theory the usage of `colorizer.greyMessage` in `prettifyObject()` could be removed, but the tests would continue to pass. 

I think I'm all set now, I'll finish this up. Thank you so much for bearing with me

**Comment by gr2m** [comment]
I think this is ready

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jsumners — APPROVED** [review]
LGTM

Should a note about the different coloring be added to the documentation for the single line option?

**Comment by mcollina** [comment]
I don't think we should add any note.

**Pull request merged**